### ISSUE TITLE
Improve Log4J assessment on vSphere

### ISSUE THREAD

**Issue by tarraschk** [wapiti-scanner/wapiti#220]
VMWare vSphere is vulnerable on the following URL:

```
https://$$IP_A_SCANNER$$/websso/SAML2/SSO/$$DOMAINE_CLIENT$$?SAMLRequest=
```

where IP_A_SCANNER must be the target's IP, and DOMAINE_CLIENT is provided by browsing the /ui/ URI when SAML is configured

This URL is not always found when targeting wapiti on `https://$$IP_A_SCANNER$$` on a vSphere.

Could we improve that, to test this URL when detecting a vSphere on the target?

Please do not hesitate to close this issue @fwininger if you think that's not "a good patch".

**Comment by fwininger** [comment]
@JulienTD can you take a look ?

**Comment by tarraschk** [comment]
To complement this: a request on https://$$IP_A_SCANNER$$/ui/ will redirect toward `https://$$IP_A_SCANNER$$/websso/SAML2/SSO/$$DOMAINE_CLIENT$$?SAMLRequest=$$SAML_TOKEN$$`, where $$SAML_TOKEN$$ **MUST** be removed from the URL to exploit the CVE as described below.

![image](https://user-images.githubusercontent.com/2057917/146751195-1ac137f0-d48f-47e7-9244-532b69e70285.png)

**Comment by JulienTD** [comment]
Ok I'm on it, but I would like to add that the second step: change the `SAML_TOKEN` to a `log4j` payload is done in the PR #214

**Comment by JulienTD** [comment]
In the Pull Request #221, I have implemented the second step which is to empty the SAMLRequest parameter and set the payload in the X-Forwarded-For header.  
But concerning the first step, which is sometimes the URL is not found, I tried to use the `wapp` module to see if we can detect the vSphere technology, but we can't unless we already are on the link `/websso/WAML2/SSO/` (example: https://vcenter.lgmserver.app/).  
Should I add in the code the `/ui/` to the `base url`  to try to detect the vSphere technology?
@fwininger
